File: 80-remove-duplicates-from-sorted-array-ii/solution.ts
// most efficient solution
export function removeDuplicates(nums: number[]): number {
  let i = 0;
  for (const num of nums) {
    if (i <= 1 || nums[i - 2] !== num) {
      nums[i] = num;
      ++i;
    }
  }

  return i;
}

File: 80-remove-duplicates-from-sorted-array-ii/solution.cpp
#include <vector>

// most efficient solution
class Solution
{
public:
  int removeDuplicates(std::vector<int> &nums)
  {
    int i = 0;
    for (const int num : nums)
    {
      if (i <= 1 || nums[i - 2] != num)
      {
        nums[i] = num;
        ++i;
      }
    }

    return i;
  }
};